### PR TITLE
added from and to to Receipt

### PR DESCRIPTION
**Pull request by ab320012**
#6948
Opening a PR to track the issue. 

I am having some issues testing this. After following initial set up for macos x I start parity and then call with the method eth_getTransactionReceipt; with any transaction it returns a null result 

curl -X POST localhost:8545 -H 'content-type: application/json'  \
-d '{
    "jsonrpc": "2.0",
    "method": "eth_getTransactionReceipt",
    "params": ["0x60a37c38327d3f3b2cace6ad89546895bf49c1fb6c70a4b23c9027967232ecb8"],
    "id": 9
}'
=>
{"jsonrpc":"2.0","result":null,"id":1}

I am thinking its a config issue not sure if anyone can assist.

### CODE REVIEW

**Comment by Tbaut** [comment]
Your node is most probably not synced. You should start parity with the `--chain dev` flag, to launch the [development chain](https://wiki.parity.io/Private-development-chain), then make a transaction (you can use [Parity UI](https://github.com/Parity-JS/shell/releases to send eth to yourself with the pre-configured account) and finally query the transaction.

**Comment by ab320012** [comment]
@Tbaut thanks for the explanation. 

Running into more issues: installed parity-ui. starting parity in the following way:
./target/release/parity --chain dev --ui-no-validation

then in parity-ui project: npm start

when i route to localhost:3000 i get a loading screen:
screen shot 2018-06-01 at 7 48 28 am

**Comment by Tbaut** [comment]
Never saw that. Could you install it with the binaries?

**Comment by ab320012** [comment]
@Tbaut , I am able to get a response, have been getting a little stuck on the implementation, hope you don't mind me querying! I am not sure where to get the 'to' property, It looks like calling action on the transaction returns an address but its not a H160. Also, having a hard time coercing the type Option<ethereum_types::H160> to Option<v1::types::hash::H160> by calling into on the ethcore object. I think I should create a coersion method in /rpc/src/v1/types/hash for the option type but want to bounce it off of someone. Again I appreciate your time, as you can see I am a bit new to rust :)

**Comment by ab320012** [comment]
commenting for gitcoin tracking - the issue is almost done just working out the failing tests

**Comment by ab320012** [comment]
@5chdn  not 100% sure what to do about this last test, it intermittently says failed. The to - from logic works as far as i can see. Is there something I am missing?

**Comment by vs77bb** [comment]
Thanks for the update @ab320012! Snoozed Gitcoin Bot 👍

**Comment by ab320012** [comment]
@5chdn all the tests pass i think this is done

**Comment by folsen** [comment]
@tomusdrw  LGTM, this doesn't seem to introduce that second DB read you were talking about, could you please give this a look?

File: ethcore/types/src/receipt.rs [hunk 1]
```diff
@@ -157,6 +157,10 @@ pub struct LocalizedReceipt {
 	pub log_bloom: Bloom,
 	/// Transaction outcome.
 	pub outcome: TransactionOutcome,
+	/// Receiver address
+	pub to: Option<H160>,
```
Comment (tomusdrw) [inline ethcore/types/src/receipt.rs]: Note we now have:
```
contract_address: Option<_>,
to: Option<_>,
```
and exactly one of the two will be `Some` for every transaction.

Just making sure that what's speced.

**Comment by folsen** [comment]
@tomusdrw Actually, I see what the problem is not, the documentation itself is really inconsistent.

There's: https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionreceipt
That says it shouldn't be included.
And there's: https://github.com/ethereum/wiki/wiki/JavaScript-API#web3ethgettransactionreceipt

That includes it, but this could be interpreted as the javascript library should be returning the fields on its own (this is obviously not how it works though).

I'm actually not sure what the right thing to do here is.

**Comment by ab320012** [comment]
Why not try to update the docs for eth get transaction receipt in ethereum

**Comment by Tbaut** [comment]
Need a PR for the [wiki](https://github.com/paritytech/wiki) too

**Pull request merged**